# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/test/webrtcproxychannel_unittest.cpp
# Commit: c43589fd0f90
# Full Hash: c43589fd0f90d174eb185b563406fd0a2e6ea712
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Update OnConnected callbacks to specify what type of proxying (if any) is being used, and implement IsProxied based on this information. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45186
# ==============================================================================

diff -r 940d7f787bea -r c43589fd0f90 media/mtransport/test/webrtcproxychannel_unittest.cpp
--- a/media/mtransport/test/webrtcproxychannel_unittest.cpp	Tue Sep 17 18:18:15 2019 +0000
+++ b/media/mtransport/test/webrtcproxychannel_unittest.cpp	Tue Sep 17 18:18:33 2019 +0000
@@ -472,7 +472,7 @@
 }
 
 void FakeWebrtcTCPSocket::InvokeOnConnected() {
-  mProxyCallbacks->OnConnected();
+  mProxyCallbacks->OnConnected(NS_LITERAL_CSTRING("http"));
 }
 
 void FakeWebrtcTCPSocket::InvokeOnRead(nsTArray<uint8_t>&& aReadData) {
@@ -494,7 +494,7 @@
 
   // WebrtcTCPSocketCallback forwards from mCallback
   void OnClose(nsresult aReason);
-  void OnConnected();
+  void OnConnected(const nsCString& aProxyType);
   void OnRead(nsTArray<uint8_t>&& aReadData);
 
   void SetUp() override;
@@ -534,7 +534,7 @@
 
   // WebrtcTCPSocketCallback
   void OnClose(nsresult aReason) override;
-  void OnConnected() override;
+  void OnConnected(const nsCString& aProxyType) override;
   void OnRead(nsTArray<uint8_t>&& aReadData) override;
 
  protected:
@@ -566,7 +566,9 @@
   AppendReadData(aReadData.Elements(), aReadData.Length());
 }
 
-void WebrtcTCPSocketTest::OnConnected() { mOnConnectedCalled = true; }
+void WebrtcTCPSocketTest::OnConnected(const nsCString& aProxyType) {
+  mOnConnectedCalled = true;
+}
 
 void WebrtcTCPSocketTest::OnClose(nsresult aReason) { mOnCloseCalled = true; }
 
@@ -614,7 +616,9 @@
   mTest->OnClose(aReason);
 }
 
-void WebrtcTCPSocketTestCallback::OnConnected() { mTest->OnConnected(); }
+void WebrtcTCPSocketTestCallback::OnConnected(const nsCString& aProxyType) {
+  mTest->OnConnected(aProxyType);
+}
 
 void WebrtcTCPSocketTestCallback::OnRead(nsTArray<uint8_t>&& aReadData) {
   mTest->OnRead(std::move(aReadData));
